app/src/main/java/com/rob729/quiethours/Fragments/MainFragment.kt

```diff
@@ -108,6 +108,37 @@ class MainFragment : Fragment() {
         setHasOptionsMenu(true)
         return binding.root
     }
+    private fun deleteAll() {
+        if (binding.emrl.visibility == View.GONE) {
+            AlertDialog.Builder(requireContext())
+                .setTitle("Delete All Profiles")
+                .setMessage("Are you sure you want to delete all the profiles?")
+                .setPositiveButton("Yes") { _, dialogInterface ->
+                    profileViewModel.allProfiles.observe(this, Observer {
```

Comment:
What is the point of observing the allProfiles inside the AlertDialog?
This isn't the expected way of getting the list of profiles.
Try out something with the profileListAdapter.

Reply:
Done